src/test/java/use_cases/add_edit_category_use_case/CategoryUCITest.java

```diff
@@ -0,0 +1,210 @@
+package use_cases.add_edit_category_use_case;
+
+import entities.EntityException;
+import entities.MonthlyStorage;
+import entities.SessionStorage;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import views.add_edit_category_views.CategoryP;
+
+import java.util.NoSuchElementException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * CategoryUCITest contains tests for the Category use cases (add & edit category).
+ */
+class CategoryUCITest {
+    private static SessionStorage session;
+
+    /**
+    Creates a SessionStorage for the following test cases.
+    */
+    @BeforeAll
+    public static void categorySetUp() {
+        session = new SessionStorage();
+    }
+
+    /**
+    Tests success add case by checking if the size of categoryData in MonthlyStorage is correctly updated.
+    */
+    @Test
+    void addCategoryInMonthSuccess() throws EntityException {
+        CategoryP presenter = new CategoryP();
+        CategoryIB interactor = new CategoryUCI(presenter);
+        MonthlyStorage monthAdd = new MonthlyStorage(6, 150);
+        session.addMonth(monthAdd);
+
+        CategoryID addID1 = new CategoryID("Salad", 12, 6, session, null);
+
+        // Check if the correct message is returned corresponding to the situation.
+        assertEquals("You have added a new category!", interactor.addCategoryInMonth(addID1).getMessage());
+        //Expected value is 2 because there is one default Category "Others" upon creation of each MonthlyStorage and one successful entry.
+        assertEquals(2, session.getMonthlyData(6).getCategoryData().size());
```

Comment:
I think you need to check whether the Category itself is added to the MonthlyStorage? 
```suggestion
        assertEquals(addID1, session.getMonthlyData(6).getCategoryData().get(1);
```
get(1) for getting Category at index 1.

Reply:
addID1 is an inputData but getCategoryData returns a Category. (I did think about doing this) I can't check for the Category object created within my use case.

Reply:
Ah, right addID1 is CategoryOD. If that's the case, can't you do
```suggestion
        assertEquals(addID1.getName(), session.getMonthlyData(6).getCategoryData().get(1).getName();
        assertEquals(addID1.getValue(), session.getMonthlyData(6).getCategoryData().get(1).getValue();
```